### ISSUE TITLE
Unexpected parsing of `not` query with Android stats

### ISSUE THREAD

**Issue by Cherry** [browserslist/browserslist#306]
If you have a `browserslist-stats.json` like the following:
```json
{
	"android": {
		"3": 0,
		"4": 0,
		"66": 3,
		"2.1": 0,
		"2.2": 0,
		"2.3": 0,
		"4.1": 0,
		"4.2-4.3": 0,
		"4.4": 0.0008008986082384436,
		"4.4.3-4.4.4": 0
	},
	"and_chr": {
		"66": 3
	},
	"firefox": {
		"57": 0.01121258051533821,
		"58": 0.0014015725644172762,
		"59": 0.04695268090797875,
		"60": 0.10922254769851773,
		"61": 5.358712475397396,
		"62": 1.956895636904607,
		"3.5": 0,
		"3.6": 0
	}
}
```
And a browserslist config like so:
```
[production]
> 0.5% in my stats
not android <= 66
not and_chr <= 66
not firefox <= 62
```
The output of `npx browserslist` is 
```
and_chr 66
android 66
```
As you expect to see, it excludes all versions of Firefox (specifically versions 62 and lower) even though they're greater than 0.5% in stats, however, Android version 66 is still being targeted.

The expected output with this example config would be no targets whatsoever.

**Comment by ai** [comment]
Why do you think that this behaviour is unexpected?

This is how I thought when I created this system. When you asked `> 0.5% in my stats, not firefox <= 62`:

1. Get all browsers `> 0.5% in my stats`
2. Next, remove all `firefox <= 62` from previous step

**Comment by Cherry** [comment]
That's exactly what I expected to happen, and it does with the `not firefox <= 62`.

`not android <= 66` and `not and_chr <= 66` don't remove the `android`/`and_chr` targets though, which, as per the previous example, I would have expected to happen.

**Comment by ai** [comment]
Got you. I asked about `android 66`.

I will look at it this week.

If you want to fix it faster, you can debug it yourself (it is simple few files project).

**Comment by Cherry** [comment]
I've found the issue:

`browserslist` is only ever aware of the latest Android version `67` (ignoring 2-4) at time of writing, and `and_chr` version `69`, as this is all that caniuse reports in its data. `caniuse` doesn't appear to report any historical data for android versions as it does with desktop browsers. As my stats are for the last 90 days or so, I'm picking up versions that caniuse and therefore browserslist isn't aware of, such as Android `66` and `and_chr` `66`.

I'm not really sure what the best solution is here.

- Should targets specified in the `browserslist-stats.json` simply be ignored if they don't exist in the latest caniuse data?
- Should there perhaps be a semver comparison when this happens?

**Comment by ai** [comment]
OH :( I hate instability of `caniuse-db` :(

I will think

**Comment by ai** [comment]
I found maybe the better way. What if we will normalize custom statistics to this `caniuse` version?

Here is a commit with normalization: 449bf6c

What do you think?

**Comment by Cherry** [comment]
This looks like a good solution. 👍

**Comment by ai** [comment]
Released in 4.3.1

**Comment by ai** [comment]
Released in 4.3.1

**Comment by ai** [comment]
Released in 4.3.1

**Comment by ai** [comment]
Released in 4.3.1
